### PR TITLE
Psp 5908 - generation for interest holders

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: source/backend/api/Areas/CompensationRequisition/Controllers/PayeeController.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(PayeeController),
+                nameof(GetPayeeById),
+                User.GetUsername(),
+                DateTime.Now);
+            _logger.LogInformation("Dispatching to service: {Service}", _compensationRequisitionService.GetType());
+
+            var pimsAcquisitionPayee = _compensationRequisitionService.GetPayeeById(id);
```
Comment (asanchezr) [inline source/backend/api/Areas/CompensationRequisition/Controllers/PayeeController.cs:57]: codecov is complaining about lack of test coverage for this class/method. Will they come at a later time?

File: source/frontend/src/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent.tsx [hunk 2]
```diff
@@ -44,7 +50,7 @@ export const ContactFilterComponent: React.FunctionComponent<
     <Formik
       enableReinitialize
       initialValues={
-        filter ?? { ...defaultFilter, searchBy: showOnlyIndividuals ? 'persons' : 'all' }
+        filter ?? { ...defaultFilter, searchBy: restrictContactType?.toString() ?? 'all' }
```
Comment (asanchezr) [inline source/frontend/src/components/contact/ContactManagerView/ContactFilterComponent/ContactFilterComponent.tsx:53]: nit: please replace hardcoded string `'all'` with corresponding enum -> `RestrictContactType.ALL`

File: source/frontend/src/features/properties/map/acquisition/add/AddAcquisitionForm.tsx [hunk 3]
```diff
@@ -204,6 +205,7 @@ export const AddAcquisitionForm = React.forwardRef<
                 <ContactInputContainer
                   field="ownerSolicitor.contact"
                   View={ContactInputView}
+                  restrictContactType={RestrictContactType.ONLY_ORGANIZATIONS}
```
Comment (asanchezr) [inline source/frontend/src/features/properties/map/acquisition/add/AddAcquisitionForm.tsx:208]: thanks for doing this. could you please add the restriction to only persons to owner representative below in this form?

File: source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateH120.ts [hunk 4]
```diff
@@ -44,20 +51,30 @@ export const useGenerateH120 = () => {
       true,
     );
     const h120CategoriesPromise = getH120Categories.execute();
+    const interestHoldersPromise = getAcquisitionInterestHolders.execute(
+      compensation.acquisitionFileId,
+    );
+    const payeePromise = getCompensationRequisitionPayee.execute(compensation.id);
```
Comment (asanchezr) [inline source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateH120.ts:57]: hmm - something doesn't add up here. Why would we call the backend "get payee by ID" passing the compensation ID...

did you mean `compensation.payees[0].id` ?

File: source/frontend/src/features/properties/map/acquisition/update/summary/UpdateAcquisitionForm.tsx [hunk 5]
```diff
@@ -262,6 +263,7 @@ const AcquisitionDetailSubForm: React.FC<{
           <ContactInputContainer
             field="ownerSolicitor.contact"
             View={ContactInputView}
+            restrictContactType={RestrictContactType.ONLY_ORGANIZATIONS}
```
Comment (asanchezr) [inline source/frontend/src/features/properties/map/acquisition/update/summary/UpdateAcquisitionForm.tsx:266]: please update ownerRep.contact to restrict to only persons

File: source/backend/api/Areas/CompensationRequisition/Controllers/PayeeController.cs (hunk 1 above)
Comment (devinleighsmith) [reply source/backend/api/Areas/CompensationRequisition/Controllers/PayeeController.cs:57]: from our best practices doc:
https://confluence.th.gov.bc.ca/display/PIMS/Best+Practices#BestPractices-UnitTests

tests do not need to be written for controller methods unless they contain non-trivial business logic or exceptions. This contains neither, so adding tests here is not required and realistically would only increase coverage and not confidence.

File: source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateH120.ts (hunk 4 above)
Comment (devinleighsmith) [reply source/frontend/src/features/properties/map/acquisition/common/GenerateForm/hooks/useGenerateH120.ts:57]: yes, will fix.

File: source/frontend/src/models/generate/acquisition/GenerateCompensationPayee.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      this.name = formatNames([
+        payee.acquisitionOwner.givenName,
+        payee.acquisitionOwner.lastNameAndCorpName,
+      ]);
+    } else if (payee?.interestHolder) {
+      if (payee.interestHolder.person) {
+        this.name = formatNames([
+          payee.interestHolder.person.firstName,
+          payee.interestHolder.person.surname,
+        ]);
```
Comment (asanchezr) [inline source/frontend/src/models/generate/acquisition/GenerateCompensationPayee.ts:23]: I noticed that in GenerateInterestHolder we use the middle-name (if present) but here we do not. Just want to confirm that is okay as per business requirements.

Also the other "persons" in this method (owner rep, owner solicitor, etc) don't have their middle-names included in the formatName() call as well.

**Review by asanchezr — APPROVED** [review]

**Pull request merged**